orca-clouddriver/src/test/java/com/netflix/spinnaker/orca/clouddriver/pipeline/providers/gce/GceDeployStagePreProcessorTest.java

```diff
@@ -0,0 +1,132 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.netflix.spinnaker.orca.clouddriver.pipeline.providers.gce;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.netflix.spinnaker.orca.clouddriver.pipeline.providers.aws.ApplySourceServerGroupCapacityStage;
+import com.netflix.spinnaker.orca.clouddriver.pipeline.providers.aws.CaptureSourceServerGroupCapacityTask;
+import com.netflix.spinnaker.orca.clouddriver.pipeline.servergroup.ResizeServerGroupStage;
+import com.netflix.spinnaker.orca.clouddriver.pipeline.servergroup.strategies.DeployStagePreProcessor.StageDefinition;
+import com.netflix.spinnaker.orca.clouddriver.pipeline.servergroup.strategies.DeployStagePreProcessor.StepDefinition;
+import com.netflix.spinnaker.orca.clouddriver.pipeline.servergroup.support.TargetServerGroup;
+import com.netflix.spinnaker.orca.clouddriver.pipeline.servergroup.support.TargetServerGroupResolver;
+import com.netflix.spinnaker.orca.pipeline.model.Stage;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class GceDeployStagePreProcessorTest {
+  private GceDeployStagePreProcessor preProcessor;
+  private ApplySourceServerGroupCapacityStage applySourceServerGroupCapacityStage;
+  private ResizeServerGroupStage resizeServerGroupStage;
+
+  @Mock private TargetServerGroupResolver targetServerGroupResolver;
+
+  @BeforeEach
+  void setUp() {
+    applySourceServerGroupCapacityStage = new ApplySourceServerGroupCapacityStage();
+    resizeServerGroupStage = new ResizeServerGroupStage();
+    preProcessor =
+        new GceDeployStagePreProcessor(
+            applySourceServerGroupCapacityStage, resizeServerGroupStage, targetServerGroupResolver);
+  }
+
+  @Test
+  void redBlackStrategyTest() {
+    when(targetServerGroupResolver.resolve(any()))
+        .thenReturn(
+            ImmutableList.of(
+                new TargetServerGroup(
+                    ImmutableMap.of("name", "testapp-v000", "zone", "us-central1-f"))));
+    Stage stage = new Stage();
+    Map<String, Object> context = new HashMap<>();
```

Comment:
There's no hard and fast rule to this, and sometimes it's difficult/impossible, but I try to make my test methods contain all the _relevant_ details, but not additional ones. In this case, this test method is testing something about the context's strategy and the rest of the context is just boilerplate to get the thing to run. So I'd be inclined to hide these details in a helper method:

`Map<String, Object> context = createContextWithStrategy(redblack);`

or maybe this is better?
```
Map<String, Object> context = createDefaultContext();
context.put("strategy", "redblack");
```

Reply:
love it, I went with option #2!